--- a/addict/addict.py
+++ b/addict/addict.py
@@ -76,15 +76,6 @@
 
     def to_dict(self):
         base = {}
-        for key, value in self.items():
-            if isinstance(value, type(self)):
-                base[key] = value.to_dict()
-            elif isinstance(value, (list, tuple)):
-                base[key] = type(value)(
-                    item.to_dict() if isinstance(item, type(self)) else
-                    item for item in value)
-            else:
-                base[key] = value
         return base
 
     def copy(self):
