--- a/addict/addict.py
+++ b/addict/addict.py
@@ -4,9 +4,6 @@
 class Dict(dict):
 
     def __init__(__self, *args, **kwargs):
-        object.__setattr__(__self, '__parent', kwargs.pop('__parent', None))
-        object.__setattr__(__self, '__key', kwargs.pop('__key', None))
-        object.__setattr__(__self, '__frozen', False)
         for arg in args:
             if not arg:
                 continue
@@ -18,9 +15,12 @@
             else:
                 for key, val in iter(arg):
                     __self[key] = __self._hook(val)
+        object.__setattr__(__self, '__frozen', False)
+        object.__setattr__(__self, '__key', kwargs.pop('__key', None))
 
         for key, val in kwargs.items():
             __self[key] = __self._hook(val)
+        object.__setattr__(__self, '__parent', kwargs.pop('__parent', None))
 
     def __setattr__(self, name, value):
         if hasattr(self.__class__, name):
