--- a/addict/addict.py
+++ b/addict/addict.py
@@ -12,11 +12,6 @@
                 continue
             elif isinstance(arg, dict):
                 for key, val in arg.items():
-                    __self[key] = __self._hook(val)
-            elif isinstance(arg, tuple) and (not isinstance(arg[0], tuple)):
-                __self[arg[0]] = __self._hook(arg[1])
-            else:
-                for key, val in iter(arg):
                     __self[key] = __self._hook(val)
 
         for key, val in kwargs.items():
