--- a/addict/addict.py
+++ b/addict/addict.py
@@ -32,8 +32,6 @@
     def __setitem__(self, name, value):
         isFrozen = (hasattr(self, '__frozen') and
                     object.__getattribute__(self, '__frozen'))
-        if isFrozen and name not in super(Dict, self).keys():
-                raise KeyError(name)
         super(Dict, self).__setitem__(name, value)
         try:
             p = object.__getattribute__(self, '__parent')
@@ -41,10 +39,6 @@
         except AttributeError:
             p = None
             key = None
-        if p is not None:
-            p[key] = self
-            object.__delattr__(self, '__parent')
-            object.__delattr__(self, '__key')
 
     def __add__(self, other):
         if not self.keys():
