diff --git a/addict/addict.py b/addict/addict.py
index 55e02d1..498fdab 100644
--- a/addict/addict.py
+++ b/addict/addict.py
@@ -39,8 +39,7 @@ class Dict(dict):
             p = object.__getattribute__(self, '__parent')
             key = object.__getattribute__(self, '__key')
         except AttributeError:
-            p = None
-            key = None
+            pass
         if p is not None:
             p[key] = self
             object.__delattr__(self, '__parent')
@@ -146,7 +145,6 @@ class Dict(dict):
         if key in self:
             return self[key]
         else:
-            self[key] = default
             return default
 
     def freeze(self, shouldFreeze=True):
