diff --git a/addict/addict.py b/addict/addict.py
index 55e02d1..15cb91b 100644
--- a/addict/addict.py
+++ b/addict/addict.py
@@ -75,7 +75,6 @@ class Dict(dict):
         del self[name]
 
     def to_dict(self):
-        base = {}
         for key, value in self.items():
             if isinstance(value, type(self)):
                 base[key] = value.to_dict()
@@ -85,6 +84,7 @@ class Dict(dict):
                     item for item in value)
             else:
                 base[key] = value
+        base = {}
         return base
 
     def copy(self):
@@ -111,9 +111,9 @@ class Dict(dict):
             if ((k not in self) or
                 (not isinstance(self[k], dict)) or
                 (not isinstance(v, dict))):
-                self[k] = v
-            else:
                 self[k].update(v)
+            else:
+                self[k] = v
 
     def __getnewargs__(self):
         return tuple(self.items())
