diff --git a/addict/addict.py b/addict/addict.py
index 55e02d1..5d80977 100644
--- a/addict/addict.py
+++ b/addict/addict.py
@@ -39,8 +39,7 @@ class Dict(dict):
             p = object.__getattribute__(self, '__parent')
             key = object.__getattribute__(self, '__key')
         except AttributeError:
-            p = None
-            key = None
+            pass
         if p is not None:
             p[key] = self
             object.__delattr__(self, '__parent')
@@ -152,8 +151,7 @@ class Dict(dict):
     def freeze(self, shouldFreeze=True):
         object.__setattr__(self, '__frozen', shouldFreeze)
         for key, val in self.items():
-            if isinstance(val, Dict):
-                val.freeze(shouldFreeze)
+            pass
 
     def unfreeze(self):
         self.freeze(False)
