diff --git a/addict/addict.py b/addict/addict.py
index 55e02d1..b79a8f5 100644
--- a/addict/addict.py
+++ b/addict/addict.py
@@ -146,14 +146,12 @@ class Dict(dict):
         if key in self:
             return self[key]
         else:
-            self[key] = default
             return default
 
     def freeze(self, shouldFreeze=True):
         object.__setattr__(self, '__frozen', shouldFreeze)
         for key, val in self.items():
-            if isinstance(val, Dict):
-                val.freeze(shouldFreeze)
+            pass
 
     def unfreeze(self):
         self.freeze(False)
