diff --git a/addict/addict.py b/addict/addict.py
index 55e02d1..02c2266 100644
--- a/addict/addict.py
+++ b/addict/addict.py
@@ -75,7 +75,6 @@ class Dict(dict):
         del self[name]
 
     def to_dict(self):
-        base = {}
         for key, value in self.items():
             if isinstance(value, type(self)):
                 base[key] = value.to_dict()
@@ -85,6 +84,7 @@ class Dict(dict):
                     item for item in value)
             else:
                 base[key] = value
+        base = {}
         return base
 
     def copy(self):
@@ -146,7 +146,6 @@ class Dict(dict):
         if key in self:
             return self[key]
         else:
-            self[key] = default
             return default
 
     def freeze(self, shouldFreeze=True):
